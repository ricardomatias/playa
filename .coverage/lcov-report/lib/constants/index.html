<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for lib/constants</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> lib/constants
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">87.5% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>56/64</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">66.67% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>4/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">33.33% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">88.89% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>56/63</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line high'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="chords.js"><a href="chords.js.html">chords.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="36" class="abs high">36/36</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="36" class="abs high">36/36</td>
	</tr>

<tr>
	<td class="file empty" data-value="index.js"><a href="index.js.html">index.js</a></td>
	<td data-value="0" class="pic empty"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	</tr>

<tr>
	<td class="file high" data-value="intervals.js"><a href="intervals.js.html">intervals.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	</tr>

<tr>
	<td class="file high" data-value="note.js"><a href="note.js.html">note.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	</tr>

<tr>
	<td class="file high" data-value="rhythmsDistributions.js"><a href="rhythmsDistributions.js.html">rhythmsDistributions.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	</tr>

<tr>
	<td class="file high" data-value="scales.js"><a href="scales.js.html">scales.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	</tr>

<tr>
	<td class="file medium" data-value="ticks.js"><a href="ticks.js.html">ticks.js</a></td>
	<td data-value="60" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 60%;"></div><div class="cover-empty" style="width:40%;"></div></div></td>
	<td data-value="60" class="pct medium">60%</td>
	<td data-value="20" class="abs medium">12/20</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="6" class="abs medium">4/6</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="6" class="abs low">2/6</td>
	<td data-value="63.16" class="pct medium">63.16%</td>
	<td data-value="19" class="abs medium">12/19</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Feb 20 2019 14:43:58 GMT+0100 (Central European Standard Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
<script src="../../block-navigation.js"></script>
</body>
</html>
