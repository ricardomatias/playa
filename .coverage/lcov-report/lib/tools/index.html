<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for lib/tools</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> lib/tools
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">91.08% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>296/325</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">85% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>102/120</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">84.75% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>50/59</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">90.76% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>275/303</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line high'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="choose.js"><a href="choose.js.html">choose.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	</tr>

<tr>
	<td class="file high" data-value="distance.js"><a href="distance.js.html">distance.js</a></td>
	<td data-value="97.37" class="pic high"><div class="chart"><div class="cover-fill" style="width: 97%;"></div><div class="cover-empty" style="width:3%;"></div></div></td>
	<td data-value="97.37" class="pct high">97.37%</td>
	<td data-value="38" class="abs high">37/38</td>
	<td data-value="94.44" class="pct high">94.44%</td>
	<td data-value="18" class="abs high">17/18</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	<td data-value="97.14" class="pct high">97.14%</td>
	<td data-value="35" class="abs high">34/35</td>
	</tr>

<tr>
	<td class="file high" data-value="distribute.js"><a href="distribute.js.html">distribute.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="24" class="abs high">24/24</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="20" class="abs high">20/20</td>
	</tr>

<tr>
	<td class="file high" data-value="euclidean.js"><a href="euclidean.js.html">euclidean.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="20" class="abs high">20/20</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="18" class="abs high">18/18</td>
	</tr>

<tr>
	<td class="file high" data-value="friendly.js"><a href="friendly.js.html">friendly.js</a></td>
	<td data-value="97.56" class="pic high"><div class="chart"><div class="cover-fill" style="width: 97%;"></div><div class="cover-empty" style="width:3%;"></div></div></td>
	<td data-value="97.56" class="pct high">97.56%</td>
	<td data-value="82" class="abs high">80/82</td>
	<td data-value="92.59" class="pct high">92.59%</td>
	<td data-value="27" class="abs high">25/27</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="13" class="abs high">13/13</td>
	<td data-value="97.4" class="pct high">97.4%</td>
	<td data-value="77" class="abs high">75/77</td>
	</tr>

<tr>
	<td class="file empty" data-value="index.js"><a href="index.js.html">index.js</a></td>
	<td data-value="0" class="pic empty"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	</tr>

<tr>
	<td class="file high" data-value="interval.js"><a href="interval.js.html">interval.js</a></td>
	<td data-value="88.89" class="pic high"><div class="chart"><div class="cover-fill" style="width: 88%;"></div><div class="cover-empty" style="width:12%;"></div></div></td>
	<td data-value="88.89" class="pct high">88.89%</td>
	<td data-value="36" class="abs high">32/36</td>
	<td data-value="85" class="pct high">85%</td>
	<td data-value="20" class="abs high">17/20</td>
	<td data-value="87.5" class="pct high">87.5%</td>
	<td data-value="8" class="abs high">7/8</td>
	<td data-value="91.43" class="pct high">91.43%</td>
	<td data-value="35" class="abs high">32/35</td>
	</tr>

<tr>
	<td class="file high" data-value="random.js"><a href="random.js.html">random.js</a></td>
	<td data-value="93.33" class="pic high"><div class="chart"><div class="cover-fill" style="width: 93%;"></div><div class="cover-empty" style="width:7%;"></div></div></td>
	<td data-value="93.33" class="pct high">93.33%</td>
	<td data-value="15" class="abs high">14/15</td>
	<td data-value="80" class="pct high">80%</td>
	<td data-value="5" class="abs high">4/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="92.86" class="pct high">92.86%</td>
	<td data-value="14" class="abs high">13/14</td>
	</tr>

<tr>
	<td class="file medium" data-value="ring.js"><a href="ring.js.html">ring.js</a></td>
	<td data-value="77.27" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 77%;"></div><div class="cover-empty" style="width:23%;"></div></div></td>
	<td data-value="77.27" class="pct medium">77.27%</td>
	<td data-value="22" class="abs medium">17/22</td>
	<td data-value="83.33" class="pct high">83.33%</td>
	<td data-value="12" class="abs high">10/12</td>
	<td data-value="40" class="pct low">40%</td>
	<td data-value="5" class="abs low">2/5</td>
	<td data-value="77.27" class="pct medium">77.27%</td>
	<td data-value="22" class="abs medium">17/22</td>
	</tr>

<tr>
	<td class="file high" data-value="roll.js"><a href="roll.js.html">roll.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="9" class="abs high">9/9</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	</tr>

<tr>
	<td class="file high" data-value="rotate.js"><a href="rotate.js.html">rotate.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	</tr>

<tr>
	<td class="file high" data-value="shuffle.js"><a href="shuffle.js.html">shuffle.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="12" class="abs high">12/12</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="9" class="abs high">9/9</td>
	</tr>

<tr>
	<td class="file medium" data-value="time.js"><a href="time.js.html">time.js</a></td>
	<td data-value="71.93" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 71%;"></div><div class="cover-empty" style="width:29%;"></div></div></td>
	<td data-value="71.93" class="pct medium">71.93%</td>
	<td data-value="57" class="abs medium">41/57</td>
	<td data-value="67.86" class="pct medium">67.86%</td>
	<td data-value="28" class="abs medium">19/28</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="10" class="abs medium">5/10</td>
	<td data-value="71.43" class="pct medium">71.43%</td>
	<td data-value="56" class="abs medium">40/56</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Feb 20 2019 15:09:45 GMT+0100 (Central European Standard Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
<script src="../../block-navigation.js"></script>
</body>
</html>
