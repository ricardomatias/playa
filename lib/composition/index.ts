export { createMotif } from './motif';
export { createChordProgression } from './chord-progression';
export { createMelodies } from './melodies';
export { createPercussion } from './percussion';
export { createArp, genHarmonicShift } from './arp';

export * as Rhythm from './rhythm';
export * from './sequence';
export * from './movement';
