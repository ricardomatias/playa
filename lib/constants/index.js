/**
 * Constants
 * @namespace {module} Constants
 */
export * from './chords';
export * from './scales';
export * from './note';
export * from './turns';
export * from './time';
export { default as MIDI_NOTES } from './midi';
export { default as TICKS } from './ticks';
export { default as INTERVALS } from './intervals';
