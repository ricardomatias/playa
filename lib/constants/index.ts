/**
 * Constants
 * @namespace {module} Constants
 */
export * from './chords';
export * from './scales';
export * from './note';
export * from './turns';
export * from './midi';
export * from './ticks';
export * from './intervals';
