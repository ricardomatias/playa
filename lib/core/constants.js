// *****************************************************************************
// SCALES
// *****************************************************************************

// TODO: REMOVE DUPLICATED SCALES
const CHROMATIC = '1P 2m 2M 3m 3M 4P 4A 5P 6m 6M 7m 7M';
const LYDIAN = '1P 2M 3M 4A 5P 6M 7M';
const MAJOR = '1P 2M 3M 4P 5P 6M 7M';
const IONIAN = '1P 2M 3M 4P 5P 6M 7M';
const MIXOLYDIAN = '1P 2M 3M 4P 5P 6M 7m';
const MINOR = '1P 2M 3m 4P 5P 6m 7m';
const AEOLIAN = '1P 2M 3m 4P 5P 6m 7m';
const DORIAN = '1P 2M 3m 4P 5P 6M 7m';
const PHRYGIAN = '1P 2m 3M 4M 5M 6m 7M';
const LOCRIAN = '1P 2m 3m 4P 5d 6m 7m';
const MAJOR_PENTATONIC = '1P 2M 3M 5P 6M';
const MINOR_PENTATONIC = '1P 3m 4P 5P 7m';
const EGYPTIAN = '1P 2M 4P 5P 7m';
const MELODIC_MINOR = '1P 2M 3m 4P 5P 6M 7M';
const ALTERED = '1P 2m 3m 3M 5d 6m 7m';
const HARMONIC_MINOR = '1P 2M 3m 4P 5P 6m 7M';

export const SCALES = {
	CHROMATIC,
	LYDIAN,
	MAJOR,
	IONIAN,
	MIXOLYDIAN,
	MINOR,
	AEOLIAN,
	DORIAN,
	PHRYGIAN,
	LOCRIAN,
	MAJOR_PENTATONIC,
	MINOR_PENTATONIC,
	EGYPTIAN,
	MELODIC_MINOR,
	ALTERED,
	HARMONIC_MINOR,
};

// *****************************************************************************
// CHORDS
// *****************************************************************************

export const CHORDS = new Map([
	[ '', [ '1P 3M 5P' ] ],
	[ 'maj', [ '1P 3M 5P' ] ],

	[ 'm', [ '1P 3m 5P' ] ],

	[ 'aug', [ '1P 3M 5A' ] ],
	[ 'dim', [ '1P 3m 5d' ] ],

	[ 'sus2', [ '1P 2M 5P' ] ],
	[ 'sus4', [ '1P 4P 5P' ] ],

	[ '5', [ '1P 5P' ] ],

	[ '7', [ '1P 3M 5P 7m', [ 'maj', '7m' ] ] ],
	[ 'm7', [ '1P 3m 5P 7m', [ 'm', '7m' ] ] ],
	[ 'm7b5', [ '1P 3m 5d 7m', [ 'dim', '7m' ] ] ],
	[ 'M7', [ '1P 3M 5P 7M', [ 'maj', '7M' ] ] ],

	[ '6', [ '1P 3M 5P 6M', [ 'maj', '6M' ] ] ],
	[ 'm6', [ '1P 3m 5P 6M', [ 'min', '6M' ] ] ],
	[ '6add9', [ '1P 3M 5P 6M 9M', [ 'maj', '6M 9M' ] ] ],

	[ '9', [ '1P 3M 5P 7m 9M', [ '7', '9M' ] ] ],
	[ 'm9', [ '1P 3m 5P 7m 9M', [ 'm7', '9M' ] ] ],
	[ 'M9', [ '1P 3M 5P 7M 9M', [ 'M7', '9M' ] ] ],

	[ '11', [ '1P 3M 5P 7m 9M 11P', [ '9', '11P' ] ] ],
	[ 'm11', [ '1P 3m 5P 7m 9M 11P', [ 'm9', '11P' ] ] ],
	[ 'M11', [ '1P 3M 5P 7M 9M 11P', [ 'M9', '11P' ] ] ],

	[ '13', [ '1P 3M 5P 7m 9M 13M', [ '11', '13M' ] ] ],
	[ 'm13', [ '1P 3m 5P 7m 9M 13M', [ 'm11', '13M' ] ] ],
	[ 'M13', [ '1P 3M 5P 7M 9M 13M', [ 'M11', '13M' ] ] ],
]);

// *****************************************************************************
// INTERVALS
// *****************************************************************************

export const INTERVALS = new Map([
	[ '1P', 0 ],
	[ '2m', 1 ], [ '2M', 2 ],
	[ '3m', 3 ], [ '3M', 4 ],
	[ '4P', 5 ], [ '4A', 6 ],
	[ '5d', 6 ], [ '5P', 7 ], [ '5A', 8 ],
	[ '6m', 8 ], [ '6M', 9 ],
	[ '7m', 10 ], [ '7M', 11 ],
	[ '8P', 12 ],
	[ '9m', 13 ], [ '9M', 14 ],
	[ '11P', 17 ], [ '11A', 18 ],
	[ '13m', 20 ], [ '13M', 21 ],
]);


// *****************************************************************************
// NOTE
// *****************************************************************************

export const SHARPS = [ 'C', 'C#', 'D', 'D#', 'E', 'F', 'F#', 'G', 'G#', 'A', 'A#', 'B' ];
export const FLATS = [ 'C', 'Db', 'D', 'Eb', 'E', 'F', 'Gb', 'G', 'Ab', 'A', 'Bb', 'B' ];

export const ENHARMONICS = [ 'C#|Db', 'D#|Eb', 'F#|Gb', 'G#|Ab', 'A#|Bb' ];

export const DIATONIC_NOTES = [ 'C', 'D', 'E', 'F', 'G', 'A', 'B' ];

