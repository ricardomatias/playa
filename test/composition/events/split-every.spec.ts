import { Events, Rhythm } from '../../../lib/composition';
import { Ticks } from '../../../lib/constants';
import { random } from '../../../lib/tools';
import '../../matchers';

describe('#splitEvery', () => {
	it('should splitEvery simple', () => {
		// given
		random.setSeed('test');

		const rhythm = Rhythm.free('4:0:0', ['1n']);

		// when
		const result = Events.splitEvery(rhythm, '1:0:0');

		// then
		expect(result[0]).toLastAround(1920);
		expect(result).toMatchInlineSnapshot(`
		[
		  [
		    {
		      "dur": 1920,
		      "isRest": false,
		      "next": 1920,
		      "time": 0,
		    },
		  ],
		  [
		    {
		      "dur": 1920,
		      "isRest": false,
		      "next": 3840,
		      "time": 1920,
		    },
		  ],
		  [
		    {
		      "dur": 1920,
		      "isRest": false,
		      "next": 5760,
		      "time": 3840,
		    },
		  ],
		  [
		    {
		      "dur": 1920,
		      "isRest": false,
		      "next": 7680,
		      "time": 5760,
		    },
		  ],
		]
	`);
	});

	it('should splitEvery 8n simple', () => {
		// given
		random.setSeed('test');

		const rhythm = Rhythm.free('1:0:0', ['8n']);

		// when
		const result = Events.splitEvery(rhythm, '4n');

		// then
		expect(result[0]).toLastAround(480);
		expect(result).toMatchInlineSnapshot(`
		[
		  [
		    {
		      "dur": 240,
		      "isRest": false,
		      "next": 240,
		      "time": 0,
		    },
		    {
		      "dur": 240,
		      "isRest": false,
		      "next": 480,
		      "time": 240,
		    },
		  ],
		  [
		    {
		      "dur": 240,
		      "isRest": false,
		      "next": 720,
		      "time": 480,
		    },
		    {
		      "dur": 240,
		      "isRest": false,
		      "next": 960,
		      "time": 720,
		    },
		  ],
		  [
		    {
		      "dur": 240,
		      "isRest": false,
		      "next": 1200,
		      "time": 960,
		    },
		    {
		      "dur": 240,
		      "isRest": false,
		      "next": 1440,
		      "time": 1200,
		    },
		  ],
		  [
		    {
		      "dur": 240,
		      "isRest": false,
		      "next": 1680,
		      "time": 1440,
		    },
		    {
		      "dur": 240,
		      "isRest": false,
		      "next": 1920,
		      "time": 1680,
		    },
		  ],
		]
	`);
	});

	it('should splitEvery complex', () => {
		// given
		random.setSeed('test');

		const rhythm = Rhythm.free('2:0:0', ['16n', '8n', '8nt', '16nt']);

		// when
		const result = Events.splitEvery(rhythm, '4n');

		// then
		expect(result[0]).toLastAround(Ticks['4n']);
		expect(result).toMatchSnapshot();
	});

	it('should splitEveryIndicies', () => {
		// given
		random.setSeed('test');

		const rhythm = Rhythm.free('1:0:0', ['8n', '4nd']);

		// when
		const pattern = Events.splitEvery(rhythm, '4n');
		const result = Events.splitEveryIndices(rhythm, '4n');
		const convert = result.map((bracket) => bracket.map((i) => rhythm[i]));

		// then
		expect(convert).toEqual(expect.arrayContaining(pattern));
	});
});
