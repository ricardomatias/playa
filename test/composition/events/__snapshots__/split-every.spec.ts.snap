// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#splitEvery should splitEvery complex 1`] = `
[
  [
    {
      "dur": 240,
      "isRest": false,
      "next": 240,
      "time": 0,
    },
    {
      "dur": 120,
      "isRest": false,
      "next": 360,
      "time": 240,
    },
    {
      "dur": 240,
      "isRest": false,
      "next": 600,
      "time": 360,
    },
  ],
  [
    {
      "dur": 160,
      "isRest": false,
      "next": 760,
      "time": 600,
    },
    {
      "dur": 160,
      "isRest": false,
      "next": 920,
      "time": 760,
    },
    {
      "dur": 160,
      "isRest": false,
      "next": 1080,
      "time": 920,
    },
  ],
  [
    {
      "dur": 160,
      "isRest": false,
      "next": 1240,
      "time": 1080,
    },
    {
      "dur": 160,
      "isRest": false,
      "next": 1400,
      "time": 1240,
    },
    {
      "dur": 240,
      "isRest": false,
      "next": 1640,
      "time": 1400,
    },
  ],
  [
    {
      "dur": 160,
      "isRest": false,
      "next": 1800,
      "time": 1640,
    },
    {
      "dur": 80,
      "isRest": false,
      "next": 1880,
      "time": 1800,
    },
    {
      "dur": 240,
      "isRest": false,
      "next": 2120,
      "time": 1880,
    },
  ],
  [
    {
      "dur": 160,
      "isRest": false,
      "next": 2280,
      "time": 2120,
    },
    {
      "dur": 240,
      "isRest": false,
      "next": 2520,
      "time": 2280,
    },
  ],
  [
    {
      "dur": 160,
      "isRest": false,
      "next": 2680,
      "time": 2520,
    },
    {
      "dur": 240,
      "isRest": false,
      "next": 2920,
      "time": 2680,
    },
  ],
  [
    {
      "dur": 160,
      "isRest": false,
      "next": 3080,
      "time": 2920,
    },
    {
      "dur": 160,
      "isRest": false,
      "next": 3240,
      "time": 3080,
    },
    {
      "dur": 240,
      "isRest": false,
      "next": 3480,
      "time": 3240,
    },
  ],
  [
    {
      "dur": 160,
      "isRest": false,
      "next": 3640,
      "time": 3480,
    },
    {
      "dur": 120,
      "isRest": false,
      "next": 3760,
      "time": 3640,
    },
    {
      "dur": 80,
      "isRest": false,
      "next": 3840,
      "time": 3760,
    },
  ],
]
`;
